scripts/ci_checks/android/macos.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+action="$1"
+
+build="darwin-x86_64"
+toolchain="$ANDROID_HOME/ndk/${CI_NDK}/toolchains/llvm/prebuilt/$build"
+
+case "$CI_ABI" in
+    "x86_64")
+        arch=$CI_ABI
+        host=$CI_ABI-linux-android
+        sysroot="$toolchain/sysroot/usr/lib/$host"
+        ;;
+    "i686")
+        arch=x86
+        host=$CI_ABI-linux-android
+        sysroot="$toolchain/sysroot/usr/lib/$host"
+        ;;
+    "aarch64")
+        arch=arm64-v8a
+        host=$CI_ABI-linux-android
+        sysroot="$toolchain/sysroot/usr/lib/$host"
+        ;;
+    "armv7a")
+        arch=armeabi-v7a
+        host=$CI_ABI-linux-androideabi
+        sysroot="$toolchain/sysroot/usr/lib/arm-linux-androideabi"
+        ;;
+    *) echo "unrecognized android ABI: '$CI_ABI'"; exit 1;;
+esac
+
+export PATH="$ANDROID_HOME/tools/bin:${PATH}"
+export PATH="$ANDROID_HOME/cmdline-tools/latest/bin:${PATH}"
+export PATH="$toolchain/bin:${PATH}"
+
+if [ "$action" == "--install" ]; then
```

Comment:
why not to use more secure bash analogues (`[[ ... ]]`) as soon as your are already using bashisms like `==` and arrays?

also, `case` fits better here than `if`, IMHO